lib/urlwatch/storage.py

```diff
@@ -600,6 +602,18 @@ def get_history_data(self, guid, count=1):
                     break
         return history
 
+    def has_history_data(self, guid):
+        if not self._cached_has_history_data_list:
+            for data in list(CacheEntry.query(self.db, CacheEntry.c.guid,
+                                              where=((CacheEntry.c.tries == 0) | (CacheEntry.c.tries == None)))):  # noqa:E711
+                self._cached_has_history_data_list.append(data[0])
+        try:
+            if self._cached_has_history_data_list.index(guid):
```

Comment:
As you're only using this to query, maybe use a [`set()`](https://docs.python.org/3/library/stdtypes.html#set) and use `.add()` to add and `guid in ...` to check membership.

Reply:
Do you mean something like this? 

```python
        if not self._cached_has_history_data_list:
            self._cached_has_history_data_list = frozenset(guid[0] for guid in list(CacheEntry.query(self.db,CacheEntry.c.guid, where=((CacheEntry.c.tries == 0) | (CacheEntry.c.tries == None))))) # noqa:E711
        for entry in self._cached_has_history_data_list:
            if entry == guid:
                return True
            else:
                continue
        return False
```